tests/FakeItEasy.Specs/CreationSpecs.cs

```diff
@@ -325,6 +325,44 @@ public void PrivateClassCannotBeFaked(Exception exception)
 "));
         }
 
+        [Scenario]
+        public void PrivateDelegateCannotBeFaked(Exception exception)
+        {
+            "Given a private delegate"
+                .See<PrivateDelegate>();
+
+            "When I create a fake of the delegate"
+                .x(() => exception = Record.Exception(this.CreateFake<PrivateDelegate>));
+
+            "Then it throws a fake creation exception"
+                .x(() => exception.Should().BeOfType<FakeCreationException>());
+
+            "And the exception message indicates the reason for failure"
+                .x(() => exception.Message.Should().Match(@"
+  Failed to create fake of type FakeItEasy.Specs.CreationSpecsBase+PrivateDelegate:
+    Can not create proxy for type FakeItEasy.Specs.CreationSpecsBase+PrivateDelegate because it is not accessible. Make it public, or internal and mark your assembly with [assembly: InternalsVisibleTo(*DynamicProxyGenAssembly2*)] attribute*
+"));
+        }
+
+        [Scenario]
+        public void PublicDelegateWithPrivateTypeArgumentCannotBeFaked(Exception exception)
+        {
+            "Given a public delegate with a private type argument"
+                .See<Func<PrivateClass>>();
+
+            "When I create a fake of the delegate"
+                .x(() => exception = Record.Exception(this.CreateFake<Func<PrivateClass>>));
+
+            "Then it throws a fake creation exception"
+                .x(() => exception.Should().BeOfType<FakeCreationException>());
+
+            "And the exception message indicates the reason for failure"
+                .x(() => exception.Message.Should().Match(@"
+  Failed to create fake of type System.Func`1[FakeItEasy.Specs.CreationSpecsBase+PrivateClass]:

```

Comment:
Ideally we'd be warning that we can't fake `Func<PrivateClass>`, as we're not actually trying to fake `PrivateClass`.
Sorry.

Reply:
Isn't it what we're doing?